src/connstring.rs

```diff
@@ -95,85 +95,91 @@ impl ConnectionString {
             options: None,
         }
     }
-}
 
-/// Parses a MongoDB connection string URI as defined by
-/// [the manual](http://docs.mongodb.org/manual/reference/connection-string/).
-pub fn parse(address: &str) -> Result<ConnectionString> {
-    if !address.starts_with(URI_SCHEME) {
-        return Err(ArgumentError(String::from("MongoDB connection string must start with \
+    /// Parses a MongoDB connection string URI as defined by
+    /// [the manual](http://docs.mongodb.org/manual/reference/connection-string/).
+    pub fn parse(address: &str) -> Result<ConnectionString> {
```

Comment:
Since I was in the process of making breaking changes, I thought it made sense to move this to  a non-instance method on ConnectionString rather than a free method.